prospero-cli/src/main/java/org/wildfly/prospero/cli/commands/channel/ChannelAddCommand.java

```diff
@@ -58,16 +69,42 @@ public ChannelAddCommand(CliConsole console, ActionFactory actionFactory) {
     public Integer call() throws Exception {
         Path installationDirectory = determineInstallationDirectory(directory);
         List<Repository> repositories;
-        repositories = RepositoryDefinition.from(repositoryDefs);
+        List<Channel> channels = null;
+            if(channelUrlOrPath.isPresent()) {
+                final ChannelCoordinate coord = ArtifactUtils.channelCoordFromString(channelUrlOrPath.get());
+                checkFileExists(coord.getUrl(), channelUrlOrPath.get());
+                ProvisioningDefinition provisioningDefinition = ProvisioningDefinition.builder()
+                        .setChannelCoordinates(channelUrlOrPath.get())
+                        .build();
```

Comment:
We could maybe add the`.setOverrideRepositories(RepositoryDefinition.from(repositoryDefs))` call to the builder, which would allow user to override repositories set in a YAML file.